src/types/Post.ts

```diff
@@ -0,0 +1,10 @@
+import { User } from './User';
+import { Comment } from './Comment';
+
+export interface Post {
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  comments: Comment[] | null;
```

Comment:
```suggestion
  comments: Comment[];
```

Метод fillter не повертає null і ти ніде не змінюєш значення масива `comments`